go.mod

```diff
@@ -1,72 +1,98 @@
 module github.com/grafana/machine-learning-go-client
 
-go 1.18
+go 1.22
```

Comment:
I'm late, but is go 1.22 really required for these upgrades? It will block projects using earlier versions of go from importing this library.

Reply:
Here's the story 😄 
- This commit added a dependency to an old version of the plugin SDK: https://github.com/grafana/machine-learning-go-client/commit/42205ca4cb339f4168accddc9b9d451824716fac
- Following that commit, I couldn't use this lib (machine-learning-go-client) in the TF provider due to import errors (it's been a while)
- Updating to the latest plugin-sdk version fixed the import but it required go 1.21, so I updated the version here.

It could be rolled-back to 1.21, but lower than that, we'll have to fix the import issues when this lib is imported in other repos

Reply:
Ahh ok, that makes sense. Thanks! I saw a PR come across that required a go version upgrade when adding this library so was wondering but 1.21 vs 1.22 doesn't seem like a big deal.